### PR TITLE
Add Open Notebook Folder functionality to Books viewlet

### PR DESCRIPTION
**Pull request by corivera**
A bunch of stuff in this PR, so here's a summary:
1. Added a button to viewlet toolbar to open a folder of Notebooks. This searches the provided directory for books and notebooks to add to the viewlet. Any notebooks that are already part of a book are not rendered as separate notebooks. There is also a Close Notebook context menu action, similar to what we have for books.
2. Changed the Open Book button to only open the book specified by the user. There was a recursive book search being done here before, but this recursive behavior was not being respected elsewhere, so trying to actually open a folder of books would fail. This recursive behavior is now provided by the new Open Notebook Folder action. The max depth config previously used for Open Book is now re-used for Open Notebook Folder.
3. Renamed some methods for better clarity.
4. Fixed an undefined error I noticed in NotebookComponent.isActive().
5. Cleaned up some of the array searching. We were using filter() in a bunch of places to search arrays, but only ever taking the first item, so I replaced these with find().
6. Changed the viewlet item tooltips to show the actual path to the book/notebook. The viewlet doesn't show any folders by design, so this helps to find the actual paths for viewlet items.

Still need to do some testing with handling special characters in the path. There's also an issue where a book can sometimes being opened twice when using Open Folder and then Open Book. Looking into these.

Capture

### CODE REVIEW

**Comment by chlafreniere** [comment]
Can we see one or more screenshots, @corivera?

**Comment by corivera** [comment]
Added to original post. Also noticed standalone notebooks still show the file extension. Looking into it.

File: extensions/notebook/src/book/bookTreeView.ts [hunk 1]
```diff
@@ -328,10 +336,57 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
...
+		let bookCollection = await this.getNotebooksInTree(folderPath);
+		for (let i = 0; i < bookCollection.bookPaths.length; i++) {
+			await this.openBook(bookCollection.bookPaths[i], undefined, false);
+		}
+		for (let i = 0; i < bookCollection.notebookPaths.length; i++) {
+			await this.openBook(bookCollection.notebookPaths[i], undefined, false, true);
+		}
+	}
+
+	private async getNotebooksInTree(folderPath: string): Promise<{ notebookPaths: string[]; bookPaths: string[]; }> {
```
Comment (chlafreniere) [inline extensions/notebook/src/book/bookTreeView.ts:369]: please make this into an interface

Comment (corivera) [reply extensions/notebook/src/book/bookTreeView.ts:369]: Which part? The method itself or the return type?

Comment (chlafreniere) [reply extensions/notebook/src/book/bookTreeView.ts:369]: return type, sorry 😄

**Pull request merged**